src/earn/EarnAddCryptoBottomSheet.tsx

```diff
@@ -0,0 +1,164 @@
+import { EarnEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import BottomSheet, { BottomSheetRefType } from 'src/components/BottomSheet'
+import Touchable from 'src/components/Touchable'
+import { CICOFlow } from 'src/fiatExchanges/utils'
+import QuickActionsAdd from 'src/icons/quick-actions/Add'
+import QuickActionsSend from 'src/icons/quick-actions/Send'
+import QuickActionsSwap from 'src/icons/quick-actions/Swap'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { isAppSwapsEnabledSelector } from 'src/navigator/selectors'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { useCashInTokens, useSwappableTokens, useTokenToLocalAmount } from 'src/tokens/hooks'
+import { TokenBalance } from 'src/tokens/slice'
+import { TokenActionName } from 'src/tokens/types'
+import { getTokenAnalyticsProps } from 'src/tokens/utils'
+
+export default function EarnAddCryptoBottomSheet({
+  forwardedRef,
+  token,
+  tokenAmount,
+}: {
+  forwardedRef: RefObject<BottomSheetRefType>
+  token: TokenBalance
+  tokenAmount: BigNumber
+}) {
+  const { t } = useTranslation()
+  const { swappableFromTokens } = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+
+  const showAdd = !!cashInTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId)
+  const showSwap =
+    isSwapEnabled &&
+    !!swappableFromTokens.find(
+      (tokenInfo) => tokenInfo.networkId === token.networkId && tokenInfo.tokenId !== token.tokenId
+    )
+  const addAmount = {
+    crypto: tokenAmount.toNumber(),
+    fiat: Math.round(
+      (useTokenToLocalAmount(tokenAmount, token.tokenId) || new BigNumber(0)).toNumber()
+    ),
+  }
+
+  const actions = [
+    {
+      name: TokenActionName.Add,
+      title: t('earnFlow.addCryptoBottomSheet.actions.add'),
+      details: t('earnFlow.addCryptoBottomSheet.actionDescriptions.add', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      }),
+      iconComponent: QuickActionsAdd,
+      onPress: () => {
+        navigate(Screens.SelectProvider, {
+          tokenId: token.tokenId,
+          flow: CICOFlow.CashIn,
+          amount: addAmount,
+        })
+      },
+      visible: showAdd,
+    },
+    {
+      name: TokenActionName.Transfer,
+      title: t('earnFlow.addCryptoBottomSheet.actions.transfer'),
+      details: t('earnFlow.addCryptoBottomSheet.actionDescriptions.transfer', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      }),
+      iconComponent: QuickActionsSend,
+      onPress: () => {
+        navigate(Screens.ExchangeQR, { flow: CICOFlow.CashIn, exchanges: [] })
```

Comment:
are exchanges supposed to be empty for the MVP?

Reply:
adding in a follow-up PR